google/cloud/doc/logging.dox

```diff
@@ -0,0 +1,93 @@
+/**
+@page logging Logging Components
+
+An overview of logging in the Google Cloud C++ client libraries.
+
+## Introduction
+
+The client libraries never use logging to report errors, but logging can be
+enabled to help troubleshoot problems when just the last error message does
+not provide a good enough indication of the root cause.
+
+In general, we abide by the following principles:
+
+- Logging should controlled by the application developer. Unless explicitly
+  instructed, the libraries produce no output to the console, except
+  to emit a message to `std::clog` immediately before a `GCP_LOG(FATAL)`
+  terminates the process.
+- Logging should have very low cost:
+  - It should be possible to disable logs at compile time, they should
+    disappear as-if there were `#%ifdef`/`#%endif` directives around them.
```

Comment:
Q: why the %'s?

Reply:
Because `#` is magic in doxygen it means "try really hard to convert the next thing to a link" and `%` is how you [disable that](https://doxygen.nl/manual/autolink.html#linkclass:~:text=If%20you%20want%20to%20prevent%20that%20a%20word%20that%20corresponds%20to%20a%20documented%20class%20is%20replaced%20by%20a%20link%20you%20should%20put%20a%20%25%20in%20front%20of%20the%20word)